src/plugin.js

```diff
@@ -173,33 +175,34 @@ const onPlayerReady = (player, options) => {
         : <div class="vjs-errors-message">${player.localize(error.message)}</div>
         </div>`;
     }
-
     display = player.errorDisplay;
-
+    // The code snippet below is to make sure we dispose any child closeButtons before
+    // making the display closeable
+    if (display.getChild('closeButton')) {
+      display.removeChild('closeButton');
+    }
+    // Make the error display closeable, and we should get a close button
+    player.errorDisplay.closeable(true);
     content.className = 'vjs-errors-dialog';
+    content.id = 'vjs-errors-dialog';
     content.innerHTML =
-      `<button class="vjs-errors-close-button"></button>
-        <div class="vjs-errors-content-container">
-          <h2 class="vjs-errors-headline">${this.localize(error.headline) }</h2>
+      `<div class="vjs-errors-content-container">
+        <h2 class="vjs-errors-headline">${this.localize(error.headline) }</h2>
           <div><b>${this.localize('Error Code')}</b>: ${(error.type || error.code)}</div>
           ${details}
         </div>
         <div class="vjs-errors-ok-button-container">
           <button class="vjs-errors-ok-button">${this.localize('OK')}</button>
         </div>`;
-
     display.fillWith(content);
-
+    // Get the close button inside the error display
+    display.contentEl().firstChild.appendChild(display.children()[0].el());
```

Comment:
I think this would be more robust as `display.getChild('closeButton').el()` instead of `display.children()[0].el()`. In case the child components change in the future.